cmd/server/main.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"log"
+	"log/slog"
+	"net"
+	"os"
+	"os/signal"
+
+	"github.com/gopheryan/jobby/internal/authinterceptors"
+	"github.com/gopheryan/jobby/internal/service"
+	grpc_recovery "github.com/grpc-ecosystem/go-grpc-middleware/recovery"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	grpc_reflection "google.golang.org/grpc/reflection"
+)
+
+const address = "localhost:8443"
+
+type UserGetterFunc func(context.Context) string
+
+func (u UserGetterFunc) GetUserContext(ctx context.Context) string {
+	return u(ctx)
+}
+
+// we have log.Fatal, but let's be consistent with slog
+func slogFatal(msg string, args ...any) {
+	slog.Error(msg, args...)
+	os.Exit(1)
+}
+
+func main() {
+
+	tlsConfig, err := NewTlsConfig()
+	if err != nil {
+		slogFatal("Failed to create TLS config", "error", err)
+	}
+	// Harcoded!
+	listener, err := net.Listen("tcp", address)
+	if err != nil {
+		slogFatal("Failed to create TLS listener", "error", err)
+	}
+	defer listener.Close()
+
+	grpcServer := grpc.NewServer(
+		grpc.ChainUnaryInterceptor(
+			grpc_recovery.UnaryServerInterceptor(),
+			authinterceptors.AuthHandlerUnaryInterceptor,
+		),
+		grpc.ChainStreamInterceptor(
+			grpc_recovery.StreamServerInterceptor(),
+			authinterceptors.AuthHandlerStreamInterceptor,
+		),
+		grpc.Creds(credentials.NewTLS(&tlsConfig)),
+	)
+
+	jobbyService := service.NewJobService(1, UserGetterFunc(authinterceptors.GetUserContext), "/tmp")
+	jobbyService.Register(grpcServer)
+
+	// So I can poke at this thing with grpcurl
+	grpc_reflection.Register(grpcServer)
+
+	signalChan := make(chan os.Signal, 1)
+	signal.Notify(signalChan, os.Interrupt)
+
+	// Catch sigterm and exit
+	go func() {
+		defer close(signalChan)
+		<-signalChan
+		slog.Info("Caught signal. Stopping Server")
+		grpcServer.Stop()
+	}()
+
+	slog.Info("Listening for gRPC requests!", "address", address)
+	err = grpcServer.Serve(listener)
+	if err != nil {
+		log.Fatalf("gRPC server returned with error: %s", err)
+	}
+
+	signal.Ignore(os.Interrupt)
+	<-signalChan
+	slog.Info("nighty night!")
+}
+
+// Hardcoded!
+func NewTlsConfig() (tls.Config, error) {
```

Comment:
https://go.dev/wiki/CodeReviewComments#initialisms

```suggestion
func NewTLSConfig() (tls.Config, error) {
```